main.go

```diff
@@ -94,6 +104,108 @@ func showErrorMessages(action string, errorMessages map[string]error) error {
 	return nil
 }
 
+type ConfProfile struct {
+	RemoteConfiguration bool `json:"active"`
+	Compliance          bool `json:"compliance"`
+	Insights            bool `json:"insights"`
+	Remediation         bool `json:"remediations"`
+}
+
+// MarshalJSON will convert the ConfProfile to a more meaningful structure
+func (p ConfProfile) MarshalJSON() ([]byte, error) {
+	type P struct {
+		RemoteConfiguration bool `json:"Remote Configuration"`
+		Compliance          bool `json:"Compliance"`
+		Insights            bool `json:"Insights"`
+		Remediation         bool `json:"Remediations"`
+	}
+	return json.Marshal(P(p))
+}
+
+// getConfProfile will retrieve the profile the system is configured
+func getConfProfile(client *http.Client) (*ConfProfile, error) {
+	var profile ConfProfile
+
+	// Get the server hostname where this host is connected
+	var serverHost, err = getBaseURL()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get server hostname: %w", err)
+	}
+
+	var uString string
+	// Check connection for satellite to create the final URL
+	base, err := url.Parse("https://" + serverHost)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get base URL: %w", err)
+	}
+
+	if serverHost != "cdn.redhat.com" && serverHost != "cdn.stage.redhat.com" {
```

Comment:
Why do check the URL of CDN? Why don't you check `server.hostname` (default value is `subscription.rhsm.redhat.com`). Is it good idea to expose addresses of internal testing servers?

Reply:
I write this in one of the previous comments https://github.com/RedHatInsights/rhc/pull/29#discussion_r1090764015